### PR TITLE
CSHARP-3899: Remove legacy driver

### PR DESCRIPTION
**Pull request by sanych-sun**

### CODE REVIEW

File: apidocs/api/index.md [hunk 1]
```diff
@@ -32,9 +32,6 @@
 <a href="xref:MongoDB.Driver"></a>
 :   [!INCLUDE[summary](../namespaces/MongoDB.Driver/MongoDB.Driver.md)]
 
-<a href="xref:MongoDB.Driver.Builders"></a>
```
Comment (sanych-sun) [inline apidocs/api/index.md:35]: There is no such namespace in current driver, it was in legacy though.

File: src/MongoDB.Driver.Core/Core/Async/AsyncQueue.cs [hunk 2]
```diff
@@ -1,99 +0,0 @@
...
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Text;
-using System.Threading;
-using System.Threading.Tasks;
-
-namespace MongoDB.Driver.Core.Async
-{
-    internal class AsyncQueue<T>
```
Comment (sanych-sun) [inline src/MongoDB.Driver.Core/Core/Async/AsyncQueue.cs:25]: Not used anymore, except of tests

File: src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs [hunk 3]
```diff
@@ -89,11 +89,6 @@ internal static BsonDocument CreateDriverDocument()
         internal static BsonDocument CreateDriverDocument(string driverVersion)
         {
             var driverName = "mongo-csharp-driver";
-            if (TryGetType("MongoDB.Driver.MongoServer, MongoDB.Driver.Legacy", out _))
```
Comment (sanych-sun) [inline src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs:92]: Decided to remove this as well, as we do not deliver the legacy driver anymore.

**Review by BorisDog — APPROVED** [review]
LGTM!
I suggest to run all variants to ensure there are not weird references. 
Also building the docs would be great.

**Review by rstam — APPROVED** [review]
LGTM

**Pull request merged**